### DISCUSSION TITLE
Panels/Scenes

### DISCUSSION THREAD

**Discussion by jens-br** [nest-desktop/nest-desktop#599] [Ideas]
Investigate the possibilities to unify the structure of the panels and scenes. It would be desirable to have a modular structure, which is identical (or at least similar) for both panels and scenes, so that new panels and scenes can be added with low effort.
`Add Panel` should give the possibility to select 2D- and 3D-panels (or at least to toggle the selection instead of the toggle on the Explorer button).
Further information: @bweyers

**Comment by babsey** [comment]
Panels
---------
Jens and me discussed the code structure of the panels. 

All panels contains `init` method to initialize panels (without data) and `update` method to update panels with activity data.
Inside of the update method it calls `updateData` for each activity component and `updateLayoutLabel` for axis labels.
